contracts/test/integration/IntegrationTestVoltMigratorRouter.t.sol

```diff
@@ -0,0 +1,162 @@
+        ERC20CompoundPCVDeposit(MainnetAddresses.COMPOUND_USDC_PCV_DEPOSIT);
+
+    ERC20CompoundPCVDeposit private compoundDaiPcvDeposit =
+        ERC20CompoundPCVDeposit(MainnetAddresses.COMPOUND_DAI_PCV_DEPOSIT);
+
+    IVolt oldVolt = IVolt(MainnetAddresses.VOLT);
+    ICore core = ICore(MainnetAddresses.CORE);
+    IERC20 private usdc = IERC20(MainnetAddresses.USDC);
+    IERC20 private dai = IERC20(MainnetAddresses.DAI);
+
+    Vm public constant vm = Vm(HEVM_ADDRESS);
+
+    uint256 public constant mintAmount = 100_000_000e18;
+
+    /// these are inverted
+    uint256 voltDaiFloorPrice = 9_000; /// 1 volt for .9 dai is the max allowable price
+    uint256 voltDaiCeilingPrice = 10_000; /// 1 volt for 1 dai is the minimum price
+
+    uint256 voltUsdcFloorPrice = 9_000e12; /// 1 volt for .9 usdc is the max allowable price
+    uint256 voltUsdcCeilingPrice = 10_000e12; /// 1 volt for 1 usdc is the minimum price
+
+    uint256 reservesThreshold = type(uint256).max; /// max uint so that surplus can never be allocated into the pcv deposit
+
+    function setUp() public {
+        // Deploy new volt token
+        newVolt = new VoltV2(MainnetAddresses.CORE);
+
+        // Deploy new PSMs
+        PegStabilityModule.OracleParams memory oracleParamsUsdc;
+        PegStabilityModule.OracleParams memory oracleParamsDai;
+
+        oracleParamsUsdc = PegStabilityModule.OracleParams({
+            coreAddress: address(core),
+            oracleAddress: address(oracle),
+            backupOracle: address(0),
+            decimalsNormalizer: 12,
+            doInvert: true,
+            volt: IVolt(address(newVolt))
+        });
+
+        oracleParamsDai = PegStabilityModule.OracleParams({
+            coreAddress: address(core),
+            oracleAddress: address(oracle),
+            backupOracle: address(0),
+            decimalsNormalizer: 0,
+            doInvert: true,
+            volt: IVolt(address(newVolt))
+        });
+
+        usdcPSM = new PriceBoundPSM(
+            voltUsdcFloorPrice,
+            voltUsdcCeilingPrice,
+            oracleParamsUsdc,
+            0,
+            0,
+            reservesThreshold,
+            10_000e18,
+            10_000_000e18,
+            IERC20(address(usdc)),
+            compoundUsdcPcvDeposit
+        );
+
+        daiPSM = new PriceBoundPSM(
+            voltDaiFloorPrice,
+            voltDaiCeilingPrice,
+            oracleParamsDai,
+            0,
+            0,
+            reservesThreshold,
+            10_000e18,
+            10_000_000e18,
+            IERC20(address(dai)),
+            compoundUsdcPcvDeposit
+        );
+
+        // Deploy Volt Migrator
+        voltMigrator = new VoltMigrator(
+            MainnetAddresses.CORE,
+            address(newVolt)
+        );
+
+        // Deploy Migrator Router
+        migratorRouter = new MigratorRouter(
+            address(newVolt),
+            address(voltMigrator),
+            daiPSM,
+            usdcPSM
+        );
```

Comment:
would way rather have the deployment and setup happen here from vip-13 instead of re-writing things.